### PR TITLE
Added hardhat and unit tests

### PR DESCRIPTION
**Pull request by omnus**
* Installed hardhat
* Moved contract into 'contracts' folder
* Added example unit tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "name": "nftdelegation",
+  "version": "1.0.0",
+  "description": "nftdelegation",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/omnus/nftdelegation.git"
```
Comment (brookr) [inline package.json:11]: If this is to be merged, it should match the target repo?

File: test/DelegationManagement.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        [1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]
+
+        const useCases = 
+        [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16]
+
+        await expect(
+          hhDelegationManagement.connect(addr1).batchDelegations(global17Times, delegationAddresses, nonExpiring, useCases, alltokens, tokenIds),
+          ).to.not.be.reverted
+
+
```
Comment (brookr) [inline test/DelegationManagement.js:99]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/omnus/nftdelegation/issues"
```
Comment (brookr) [inline package.json:16]: same

File: package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  "homepage": "https://github.com/omnus/nftdelegation#readme",
```
Comment (brookr) [inline package.json:18]: same

File: package.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "nftdelegation",
+  "version": "1.0.0",
```
Comment (brookr) [inline package.json:3]: Would love to see SEMVER, but this project doesn't seem to be using it... yet?

File: .gitignore [hunk 6]
```diff
@@ -1 +1,4 @@
 .DS_Store
+node_modules
+cache
+artifacts
```
Comment (brookr) [inline .gitignore:4]:
```suggestion
artifacts

```

Comment (brookr) [reply .gitignore:4]: ...just to get rid of the "no newline at end of file" indicator in the PR :P

**Review by brookr — CHANGES_REQUESTED** [review]
Thanks for this! Would love to see this project integrate some tests! I don't think this is quite ready for direct merge, very minor notes in the review.

**Comment by a2rocket** [comment]
outdated PR

**Pull request closed without merging**